gateway/ipfs.go

```diff
@@ -88,15 +88,25 @@ func (g *Gateway) renderIPFSPath(c *gin.Context, base, pth string) {
 	}
 	defer f.Close()
 
```

Comment:
A bit weird diff, but this is the same code as pre-review, just that the content-type detection logic was extracted to a new `detectReaderContentType`, that now is also leveraged by `gateway/buckets.go`, so no repeated code.